Better ray example app & doc refresh

jieru-hu opened the pull request:
<!-- Thank you for sending a PR and taking the time to improve Hydra -->

## Motivation
A refresh of the example app and the documentation.


### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

Run the ray example app locally.  For now the example app is not included in integration tests, this is due to our test user's very limited permission.
CI should pass.
## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

shagunsodhani commented on website/docs/plugins/ray_launcher.md:
```diff
@@ -3,108 +3,51 @@ id: ray_launcher
 
 :::caution
-This plugins depends on `cloudpickle`, as a result please make sure your local machine and the AWS cluster runs the same version of: `ray`, `hydra-core` and `python`. You can install the desired software version by overriding `RayClusterConf.setup_commands`.
+`ray autoscaler` expects your AWS credentials has certain permissions for [`EC2`](https://aws.amazon.com/ec2) and [`IAM`](https://aws.amazon.com/iam). This [ray issue](https://github.com/ray-project/ray/issues/9327) provides some contexts.
```
```suggestion
`ray autoscaler` expects your AWS credentials have certain permissions for [`EC2`](https://aws.amazon.com/ec2) and [`IAM`](https://aws.amazon.com/iam). This [ray issue](https://github.com/ray-project/ray/issues/9327) provides some more context.
```

omry replied:
"this issue" sounds like there is a problem.
"read this for more information"?

shagunsodhani reviewed (commented):
Skimmed through the commit. Will set it up locally in a day or two and will provide more comments then.

omry commented on plugins/hydra_ray_launcher/examples/multi_module/train.py:
```diff
@@ -8,7 +8,7 @@
 log = logging.getLogger(__name__)
 
 
-@hydra.main(config_path="conf", config_name="config")
```
Please place your configs in a conf directory.
There are some nasty side effects to not having a config dir: anything under your Python file becomes eligible to be imported as a config. any directory with yaml files becomes a config group etc. this is not good practice.

omry commented on plugins/hydra_ray_launcher/examples/multi_module/config.yaml:
```diff
@@ -1,4 +1,9 @@
-# @package _global_
+defaults:
```
what does this config has to do with multi_module?

omry commented on website/docs/plugins/ray_launcher.md:
```diff
@@ -3,108 +3,51 @@ id: ray_launcher
+`ray_aws` launcher is built on top of ray's [autoscaler cli](https://docs.ray.io/en/latest/autoscaling.html). To get started, you need to 
+[config your AWS credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html).
 
 :::caution
```
:::important?
:::note?

omry commented on website/docs/plugins/ray_launcher.md:
```diff
@@ -3,108 +3,51 @@ id: ray_launcher
 :::
 
-Now we can go ahead and run `train.py` using `ray_aws` launcher
+`ray autoscaler` expects a yaml file to provide configuration for the EC2 cluster; we've schematized the configs in [`hydra_ray_launcher.hydra_plugins.hydra_ray_launcher.conf.__init__.RayClusterConf`](https://github.com/facebookresearch/hydra/blob/master/plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/conf/__init__.py), 
```
`__init__` is not a part of the module to the conf.
(try to import that thing, it will not work).

Just give the class name with a link.

jieru-hu commented:
@omry @shagunsodhani 
thx for the comments. Response down inline.

>Skimmed through the commit. Will set it up locally in a day or two and will provide more comments then.

Sure. If you are interested in running the examples and need credentials, pls ping me so I can create an AWS user for you :) 


>>Please place your configs in a conf directory.
There are some nasty side effects to not having a config dir: anything under your Python file becomes eligible to be imported as a config. any directory with yaml files becomes a config group etc. this is not good practice.

Sounds good! Should we document this somewhere in our doc? (I don't think I've seen it mentioned in the doc.)

>what does this config has to do with multi_module?

renamed it to `upload_download`, probably a more reasonable name since the example focuses on that.

>:::important?
:::note?

important it is then :) 


>"this issue" sounds like there is a problem.
"read this for more information"?

Fixed.

>__init__ is not a part of the module to the conf.
(try to import that thing, it will not work).
Just give the class name with a link.

done!

omry reviewed (approved):
lgtm
